components/datepicker/DatePickerInputRange.js

```diff
@@ -0,0 +1,298 @@
+import React, { Fragment, PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import Box from '../box';
+import DayPickerInput from 'react-day-picker/DayPickerInput';
+import Icon from '../icon';
+import InputMetaPropTypes from '../input/InputMetaPropTypes';
+import NavigationBar from './NavigationBar';
+import WeekDay from './WeekDay';
+import { convertModifiersToClassnames, isSelectingFirstDay } from './utils';
+import { DateUtils } from 'react-day-picker/lib/src/index';
+import { IconCalendarSmallOutline, IconWarningBadgedSmallFilled } from '@teamleader/ui-icons';
+import { TextSmall } from '../typography';
+import cx from 'classnames';
+import omit from 'lodash.omit';
+import theme from './theme.css';
+
+class DatePickerInputRange extends PureComponent {
+  state = {
+    selectedStartDate: null,
+    selectedEndDate: null,
+    mouseEnteredEndDate: null,
+    inputHasFocus: false,
+  };
+
+  static getDerivedStateFromProps(props, state) {
+    if (
+      props.selectedRange !== undefined &&
+      (props.selectedRange.selectedStartDate !== state.selectedStartDate ||
+        props.selectedRange.selectedEndDate !== state.selectedEndDate)
+    ) {
+      return {
+        selectedStartDate: props.selectedRange.selectedStartDate,
+        selectedEndDate: props.selectedRange.selectedEndDate,
+        mouseEnteredEndDate: props.selectedRange.selectedEndDate,
+      };
+    }
+
+    return null;
+  }
+
+  componentWillUnmount() {
+    clearTimeout(this.timeout);
+  }
+
+  focusFrom = () => {
+    this.timeout = setTimeout(() => this.startDate.getInput().focus(), 100);
+  };
+
+  focusTo = () => {
+    this.timeout = setTimeout(() => this.endDate.getInput().focus(), 0);
```

Comment:
Why do we need the `setTimeout` here before focussing the input fields?

Reply:
This is because of the third-party `react-day-picker` component that we are wrapping.
Check the comment above the `focusTo` method in their `Range with two inputs` example: http://react-day-picker.js.org/examples/input-from-to
This is due to some cross-browser issues they're explaining above the `handleInputBlur` and `handleOverlayBlur` methods: https://github.com/gpbl/react-day-picker/blob/master/src/DayPickerInput.js